### DISCUSSION TITLE
PostgreSQL - Support for other schemas (not only public)

### DISCUSSION THREAD

**Discussion by RenanStigliani** [directus/directus#3164] [Archive]
Hi, I'm testing Directus for the first time.

I just hooked it up to my PostgreSQL database but Directus only lists my tables under the public schema on the collections page.

I couldn't find anything on how to setup other schemas... Is it currently supported?

**Comment by benhaynes** [comment]
I don't believe so, to get proper database abstraction across all vendors we need to focus support on those features that are also shared across all vendors. There has been interest in adding support for other postgres schemas, but that's not likely to be implemented soon unless we get a pull request or it becomes sponsored.

**Reply by RenanStigliani** [reply]
Ok. MS SQL also supports schemas, so adding support to it could help more than one vendor. Can you point me to where in the code the collections discoverability is done? I could look into submitting a PR depending on the complexity

**Comment by MiniDigger** [comment]
Uuhm, Ben, this wrong. I (and others) worked with rijk on getting this done and it works pretty great in the latest prereleases :)
The key is setting a search path via the env vars

**Reply by peteruithoven** [reply]
Could you share and example? I'm seeing multiple people confused on how to add multiple.

**Reply by gdcl** [reply]
yes please - getting error message to use array syntax, but array syntax also not workingDetected comma in searchPath :
"public,content_coverage".If you are trying to specify multiple schemas, use Array syntax: ['public', 'content_coverage']
but then:
 Detected comma in searchPath "['public','content_coverage']".If you are trying to specify multiple schemas, use Array syntax: ['['public'', ''content_coverage']']

Reference to correct syntax would be appreciated

**Reply by rijkvanzanten** [reply]
> Reference to correct syntax would be appreciated

https://docs.directus.io/reference/environment-variables/#environment-syntax-prefix

Keep in mind that Directus doesn't natively support / do anything with schemas. That env var is just passed on to the underlying `pg` library connection. Your mileage may vary when you add multiple schemas simultaneously 👍🏻

**Reply by productdevbook** [reply]
@rijkvanzanten  404 url

**Reply by MiniDigger** [reply]
@productdevbook here you go https://docs.directus.io/self-hosted/config-options/#environment-syntax-prefix

**Reply by productdevbook** [reply]
@MiniDigger thank you, where schema path env ?

**Comment by oreilles** [comment]
I'm pretty sure a few weeks ago that I saw on the docs that you could specify the schemas to load into directus in the .env file, but I couldn't find it anymore. What's the current situation on this ?

**Reply by MiniDigger** [reply]
the env var to set the schema is "DB_SEARCH_PATH". I haven't encountered any issues anymore the last months.

**Reply by oreilles** [reply]
Is it possible to add multiple schemas to search in ? Directus complains with "Detected comma in searchPath".

**Reply by tv42** [reply]
This is the correct syntax for `.env`: `DB_SEARCH_PATH="array:foo,bar"`

But that just causes directus to crash when it introspects the schema at startup.

Having actual support for this basic, fundamental, feature of Postgres is a dealkiller :-(

**Reply by alexdeia** [reply]
```DB_SEARCH_PATH="array:public,another-scheme"```
it's works, thanks!

**Reply by 1011382654** [reply]
> `DB_SEARCH_PATH="array:public,another-scheme"` 它的作品，谢谢！

Why does the document have no DB_ SEARCH_ PATH
![image](https://user-images.githubusercontent.com/26688758/205884049-c4559e5c-7f87-438f-8f87-b3ae39cfe8c8.png)

**Reply by MiniDigger** [reply]
directus doesnt document all possible db options, it only links to where you can find the rest, as they are provided by a library: https://docs.directus.io/self-hosted/config-options.html#database  
https://knexjs.org/guide/#configuration-options

**Reply by rijkvanzanten** [reply]
> directus doesnt document all possible db options, it only links to where you can find the rest, as they are provided by a library: https://docs.directus.io/self-hosted/config-options.html#database https://knexjs.org/guide/#configuration-options

As per the docs 🙂 :

> **Additional Database Variables**
> 
> All DB_* environment variables are passed to the connection configuration of a [Knex instance](http://knexjs.org/). Based on your project's needs, you can extend the DB_* environment variables with any config you need to pass to the database instance.
